fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -574,6 +574,50 @@ class ApplyMetadataMutationsImpl {
 		    .detail("LogRangeEnd", logRangeEnd);
 	}
 
+	void checkSetConstructKeys(MutationRef m) {
+		if (!SERVER_KNOBS->GENERATE_DATA_ENABLED) {
+			return;
+		}
+		if (!m.param1.startsWith(globalKeysPrefix)) {
+			return;
+		}
+		if (!toCommit) {
+			return;
+		}
+
+		if (m.param1.startsWith(constructDataKey)) {
+			// std::string s = "\xcf\xdf";
+			// Value v = encodeConstructValue(s, 100, 0x2ff, 123);
+			// TraceEvent(SevDebug,"ConstructDataDebug").detail("V", v);
+
+			uint64_t valSize, keyCount, seed;
+			Standalone<StringRef> prefix;
+			std::tie(prefix, valSize, keyCount, seed) = decodeConstructKeys(m.param2);
+			if (keyCount >= UINT16_MAX || valSize >= CLIENT_KNOBS->VALUE_SIZE_LIMIT) {
```

Comment:
what's the thinking of this limit `UINT16_MAX`? Related to shard size?

Reply:
I did not wish to add another knob, but wanted to pick some limit. I did not want to limit the number of keys requested to a single shard. `UINT16_MAX` is already defined so is convenient and avoids cluttering up things with new knobs.

It probably would make sense to limit the number of injected mutations per version, so if a large `keyCount` is requested we do not somehow exceed the 5 Sec limit. I will add that.